examples/fluids/qfunctions/stabilization.h

```diff
@@ -0,0 +1,167 @@
+// Copyright (c) 2017-2022, Lawrence Livermore National Security, LLC and other CEED contributors.
+// All Rights Reserved. See the top-level LICENSE and NOTICE files for details.
+//
+// SPDX-License-Identifier: BSD-2-Clause
+//
+// This file is part of CEED:  http://github.com/ceed
+
+/// @file
+/// Helper functions for computing stabilization terms of a newtonian simulation
+
+
+#ifndef stabilization_h
+#define stabilization_h
+
+// *****************************************************************************
+// Helper function for computing the variation in primitive variables,
+//   given Tau_d
+// *****************************************************************************
+CEED_QFUNCTION_HELPER void dYFromTau(CeedScalar Y[5], CeedScalar Tau_d[3],
+                                     CeedScalar dY[5]) {
+  dY[0] = Tau_d[0] * Y[0];
+  dY[1] = Tau_d[1] * Y[1];
+  dY[2] = Tau_d[1] * Y[2];
+  dY[3] = Tau_d[1] * Y[3];
+  dY[4] = Tau_d[2] * Y[4];
+}
+
+// *****************************************************************************
+// Helper function for computing the stabilization terms
+// *****************************************************************************
+CEED_QFUNCTION_HELPER void Stabilization(NewtonianIdealGasContext gas, State s,
+    CeedScalar Tau_d[3], CeedScalar Y[5], const CeedScalar x[3],
+    CeedScalar stab[5][3]) {
+  CeedScalar dY[5];
+  const CeedScalar dx_i[3] = {0};
+  StateConservative dF[3];
+  dYFromTau(Y, Tau_d, dY);
+  State ds = StateFromY_fwd(gas, s, dY, x, dx_i);
+  FluxInviscid_fwd(gas, s, ds, dF);
+  for (CeedInt j=0; j<3; j++) {
+    CeedScalar dF_j[5];
+    UnpackState_U(dF[j], dF_j);
+    for (CeedInt k=0; k<5; k++)
+      stab[k][j] += dF_j[k];
```

Comment:
Should we move the zeroing of `stab[][]` into the function itself. Seems like a simple thing that might prevent a headache in the future.

Reply:
Good call!